fe/fe-core/src/main/java/org/apache/doris/nereids/util/TypeCoercionUtils.java

```diff
@@ -189,6 +193,10 @@ public static Optional<DataType> findTightestCommonType(DataType left, DataType
             }
         } else if (left instanceof CharacterType || right instanceof CharacterType) {
             tightestCommonType = StringType.INSTANCE;
+        } else if (left instanceof DecimalType && right instanceof IntegralType) {
+            tightestCommonType = DecimalType.forType(right);
```

Comment:
What about this case:
 - left: Decimal(20,0)
 - right: int
Here the tightestCommonType is Decimal(10, 0), which cannot cover left.
Is this a problem?

Reply:
yes, should use 
```
DecimalType.widerDecimalType(left, DecimalType.forType(right));
```

Reply:
it would be better to add this case as a UT

Reply:
you are right, thx pro

Reply:
you are right, thx bro